### PR TITLE
Remove core 1 layer in graph

### PR DESCRIPTION
**Pull request by ptajvar**
## Description

Please include a summary of the change and which issue(s) have been fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Core-1 layer nodes in the megacluster don't add any information that is needed for resolving the components in the graph step as they cannot contain any crossing edges. Given that they have a significant share of nodes (20-30%), removing them can lower the memory usage and runtime of the community detection methods (FLP, Leiden, and refinements). We have therefore decided to peel off the core-1 layer before starting community detection and then reattach them after resolving components (cells).

Fixes: PNA-3603

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested on samples with larger core-1 layer to confirm the speed up and output equivalence.

## PR checklist:

- [X] This comment contains a description of changes (with reason).
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If a new tool or package is included, I have updated dependencies in `pyproject.toml` and [cited it properly](../CITATIONS.md)
- [ ] I have checked my code and documentation and corrected any misspellings
- [ ] I have documented any significant changes to the code in [CHANGELOG.md](../CHANGELOG.md)

---

> [!NOTE]
> **Medium Risk**
> Changes the core graph component-finding pipeline and which edges are kept or dropped during absorption, which can shift cell boundaries and downstream PXL outputs compared to prior runs.
> 
> **Overview**
> The PNA **graph** step now **strips the core-1 layer** (leaf UMIs outside the 2-core) before fast label propagation and Leiden, runs community detection on the smaller 2-core edgelist, then **reattaches** peeled edges in an iterative **`absorb_core1_layer`** phase. Reabsorption assigns frontier edges to a single known component, drops conflicts, and treats ambiguous “fused” bridges that would merge components.
> 
> **`--core1-absorption-max-iterations`** (default **5**) caps absorption rounds and is wired through the CLI, **`find_components`**, and **`build_pxl_file_with_components`**. **`peel_core1_nodes`** and **`absorb_core1_layer`** live in **`component_recovery_utils`**.
> 
> Graph **`report.json`** gains core-1 peel/absorption counts and **`fraction_core1_edges_reabsorbed`**. Changelog and unit tests cover peeling chains, absorption edge cases, and report snapshots.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 86b1134fd2ed50b13f0c577a9944258db8e61088. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/pixelator/pna/graph/community_detection.py [hunk 1]
```diff
@@ -500,14 +506,21 @@ def find_components(
...
+
     logger.info("Running FLP + Leiden native step")
     (
         partitioned_edgelist_path,
         pre_recovery_stats,
         post_flp_stats,
         post_recovery_stats,
     ) = run_hybrid_community_detection(
-        parquet_file=str(filtered_edgelist_path),
+        parquet_file=str(core2_edgelist_path),
```
[the remark this answers is not in the document]
Comment (ptajvar) [reply src/pixelator/pna/graph/community_detection.py:523]: This is true but I wonder if should actually do something about it. A component with fewer 8000 UMIs with core number higher than 1 are very small.

[the remark this answers is not in the document]
Comment (ptajvar) [reply src/pixelator/pna/graph/community_detection.py]: Considering it a bit more, I think we should fix this anyway because it can be misleading if the size limit is only applied to core>1 and might result in unexpected outcomes for atypical samples. Given that we want to avoid large components going into refinement, we can do a max-size-only filtering here and then a filtering for both min and max sizes in the very end. I will push a fix for this.

[the remark this answers is not in the document]
Comment (ptajvar) [reply src/pixelator/pna/graph/community_detection.py:523]: We discussed this a little bit and decided to keep it as it is for now. The issue is that during refinement, we don't have access to a component's full size (including it's core-1 nodes) unless we reattach it which would defeat the purpose of peeling core-1. Two alternatives would be to either, not discard tiny components at all during refinement, or discard tiny components with a lower threshold compared to MIN_PNA_COMPONENT_SIZE. The former would make the process very slow, going through all small components. The latter would introduce a new parameter. Therefore, for now, we decided to keep it as it is and later consider adding introducing another threshold for refinement (pruning) that is lower than the final filtering lower threshold (MIN_PNA_COMPONENT_SIZE by default).

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### Added
+- The PNA graph step now peels the core-1 layer off the edgelist before fast label propagation and Leiden,
+  then reattaches it to the resolved components in an iterative absorption phase (capped at 5 iterations
+  by default, configurable via `--core1-absorption-max-iterations`).
```
Comment (johandahlberg) [inline CHANGELOG.md:13]:
```suggestion
- The PNA graph step now peels the core-1 layer off the edgelist before fast label propagation and Leiden,
  then reattaches it to the resolved components in an iterative absorption phase (capped at 5 iterations
  by default, configurable via `--core1-absorption-max-iterations`). This improves runtime by 2x and lower memory usage in some scenarios.
```

**Review by johandahlberg — APPROVED** [review]
Small comment on the changelog, otherwise this lgtm.
